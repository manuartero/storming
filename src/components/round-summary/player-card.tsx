import { Avatar } from "./avatar/avatar";
import c from "classnames";

import styles from "./player-card.module.css";

type Props = {
  player: PlayerStatus;
  active?: boolean;
  clickable?: boolean;
  onClick: (player: PlayerStatus) => void;
};

export function PlayerSummary({ player, active, clickable, onClick }: Props) {
  return (
    <article
      className={c(
        styles.playerCard,
        styles[player.player],
        clickable && styles.clickable,
        active && styles.active
      )}
      role="button"
      aria-label={`${player.player} summary`}
      aria-disabled={!clickable}
      aria-roledescription="player summary"
      onClick={() => clickable && onClick(player)}
    >
      <div className={styles.playerPoints}>
        <Avatar player={player.player} />
        <span className={styles.points}>{player.points}p</span>
      </div>
      <div className={styles.extraPoints}></div>
    </article>
  );
}
