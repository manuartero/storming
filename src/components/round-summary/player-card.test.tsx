import { render, screen } from "@testing-library/react";
import { PlayerCard } from "./player-card";

describe("<PlayerCard />", () => {
  it("render: article role", () => {
    const playerStatus = {
      player: "player" as const,
      points: 3,
      greatestEmpirePoint: false,
    };
    render(<PlayerCard player={playerStatus} onClick={() => {}} />);

    const playerCard = screen.getByRole("article", { name: "player summary" });
    expect(playerCard).toMatchSnapshot();
  });
});
