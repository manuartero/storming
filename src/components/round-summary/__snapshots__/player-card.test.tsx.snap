// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<PlayerCard /> render: article role 1`] = `
<article
  aria-disabled="true"
  aria-label="player summary"
  aria-roledescription="player summary"
  class="playerCard player"
>
  <div
    class="playerPoints"
  >
    <div
      aria-label="player avatar"
      aria-roledescription="game avatar"
      class="avatar player"
      role="img"
    />
    <span
      class="points"
    >
      3
      p
    </span>
  </div>
  <div
    class="extraPoints"
  />
</article>
`;
